packages/api/tests/endpoints/async-operations/test-create-async-operations.js

```diff
@@ -0,0 +1,480 @@
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(asyncOperation)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test('POST returns a 404 if the requested path does not exist', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+  const response = await request(app)
+    .post(`/asyncOperations/${asyncOperation.id}`)
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  t.is(response.statusCode, 404);
+});
+
+test('POST returns a 400 response if invalid JSON provided', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const response = await request(app)
+    .post('/asyncOperations')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send('asdf');
+
+  const error = JSON.parse(response.error.text);
+
+  t.is(response.statusCode, 400);
+  t.is(error.error, 'Bad Request');
+  t.is(error.message, 'Async Operations require an ID');
+});
+
+test('post() does not write to PostgreSQL/Elasticsearch if writing to DynamoDB fails', async (t) => {
+  const { testKnex } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeAsyncOperationModel = {
+    create: () => {
+      throw new Error('something bad');
+    },
+    exists: () => Promise.resolve(false),
+    delete: () => Promise.resolve(true),
+  };
+
+  const expressRequest = {
+    body: asyncOperation,
+    testContext: {
+      knex: testKnex,
+      asyncOperationModel: fakeAsyncOperationModel,
+    },
+  };
+
+  const response = buildFakeExpressResponse();
+
+  await post(expressRequest, response);
+
+  t.true(response.boom.badImplementation.calledWithMatch('something bad'));
+
+  t.false(await t.context.esAsyncOperationsClient.exists(
+    asyncOperation.id
+  ));
+  t.false(
+    await t.context.asyncOperationPgModel.exists(t.context.testKnex, {
+      id: asyncOperation.id,
+    })
+  );
+});
+
+test('post() does not write to DynamoDB/Elasticsearch if writing to PostgreSQL fails', async (t) => {
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeAsyncOperationPgModel = {
+    create: () => Promise.reject(new Error('something bad')),
+  };
+
+  const fakeAsyncOperationModel = {
```

Comment:
Why are we mocking this model? In general I'd prefer to only mock one of the models per test so that we're only mocking one branch of the logic and not multiple. Otherwise the test could be prone to unexpected passes or failures

Reply:
[`7341e50` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/7341e50c4cad294e1c53ae68a8dd5bf599f6568d)